### ISSUE TITLE
UIWebView method - (NSString*)stringByEvaluatingJavaScriptFromString:(NSString*)string throws exception

### ISSUE THREAD

**Issue by Juju4ka1** [Microsoft/WinObjC#988]
Hi.
If you create UIWebView and load simple html page with simple javascript in it and then try to invoke this script from native using 'stringByEvaluatingJavaScriptFromString:' method, you will get 'UNSException' exception.

For instance: (simple html page was taken from example https://code.msdn.microsoft.com/windowsapps/XAML-WebView-control-sample-58ad63f7)

```
 self.webView = [[UIWebView alloc] initWithFrame:self.view.frame];
    self.webView.delegate = self;
NSString* appHtml = @"<html><head><script type='text/javascript'>function doSomething(){  document.getElementById('myDiv').innerText = 'GoodBye';return 'Hello World!'; }</script></head><body><div id='myDiv'>Hello</div></body></html>";   
[self.webView loadHTMLString:appHtml baseURL:nil];
[self.view addSubview:_webView];

.....
//Invoke script when webView finished loading
- (void)webViewDidFinishLoad:(UIWebView *)webView {
[_webView stringByEvaluatingJavaScriptFromString:@"doSomething"]; //You will get exception here
}
```

**Comment by jboich** [comment]
Hi, there.

As a hack around solution to get this feature working, we added method 
```
- (void)evaluateJavaScript:(NSString *)javaScriptString 
         completionHandler:(void (^)(id, NSError *error))completionHandler;
```
to UIWebView class and use [_xamlWebControl.InvokeScriptAsync](https://msdn.microsoft.com/en-us/library/windows.ui.xaml.controls.webview.invokescriptasync.aspx) in implementation.

**Comment by DHowett-MSFT** [comment]
@jboich Awesome. Have you run into any compatibility issues?

**Comment by Juju4ka1** [comment]
Nope, because we wrapped code using preprocessor directive. So objective-c code is:
```
#if !WINOBJC
    NSString *result = [self.webView stringByEvaluatingJavaScriptFromString:javascriptCommand];
    if (completionHandler) {
        completionHandler(result,nil);
    }
#else
    [self.webView evaluateJavaScript:javascriptCommand completionHandler:completionHandler];
#endif
```
where webView is an instance of UIWebView class.

**Comment by jaredhms** [comment]
Looks like a dupe of #582.  

The workaround appears viable, but we should probably also stub out or properly fix stringByEvaluatingJavaScriptFromString.  The documentation on [WebView.InvokeScript](https://msdn.microsoft.com/en-us/library/windows/apps/windows.ui.xaml.controls.webview.invokescript.aspx) is a bit light, so it's not clear why it's throwing an exception.  I suspect it's because it needs to wait/block on the UIThread, which isn't supported in UWPs, but it would be helpful to confirm the type of C++ exception being thrown by invokeScript.

@jboich, @Juju4ka1; do you plan to send a PR for this?

**Comment by jboich** [comment]
Hi, @jaredhms. Thanks for your feedback.  Yes, we will submit pull request.
Best regards, Julia

**Comment by tadam-msft** [comment]
Fixed by #1638
